src/main/java/pl/edu/pw/mini/ingreedio/api/ingredient/service/IngredientService.java

```diff
@@ -57,88 +53,44 @@ public Ingredient addIngredient(Ingredient ingredient) {
-            .getUserByUsername(authService.getCurrentUsername());
-        if (userOptional.isPresent()) {
-            User user = userOptional.get();
-            return user.getAllergens().stream().map(ingredientDtoMapper).toList();
-        }
-        return List.of();
+    public List<Ingredient> getAllergens(User user) {
+        return user.getAllergens().stream().toList();
     }
 
     @Transactional
-    public boolean likeIngredient(Long id) {
-        Optional<User> userOptional = userService
-            .getUserByUsername(authService.getCurrentUsername());
-        if (userOptional.isPresent()) {
-            User user = userOptional.get();
-            Optional<Ingredient> ingredientOptional = ingredientRepository.findById(id);
-            if (ingredientOptional.isPresent()) {
-                Ingredient ingredient = ingredientOptional.get();
-                userService.likeIngredient(user, ingredient);
-                return true;
-            }
-        }
-        return false;
+    public void likeIngredient(long id, User user) throws IngredientNotFoundException {
+        Ingredient ingredient = ingredientRepository.findById(id)
+            .orElseThrow(() -> new IngredientNotFoundException(id));
+
+        userService.likeIngredient(user, ingredient);
     }
 
     @Transactional
-    public boolean unlikeIngredient(Long id) {
-        Optional<User> userOptional = userService
-            .getUserByUsername(authService.getCurrentUsername());
-        if (userOptional.isPresent()) {
-            User user = userOptional.get();
-            Optional<Ingredient> ingredientOptional = ingredientRepository.findById(id);
-            if (ingredientOptional.isPresent()) {
-                Ingredient ingredient = ingredientOptional.get();
-                userService.unlikeIngredient(user, ingredient);
-                return true;
-            }
-        }
-        return false;
+    public void unlikeIngredient(long id, User user) throws IngredientNotFoundException {
+        Ingredient ingredient = ingredientRepository.findById(id)
+            .orElseThrow(() -> new IngredientNotFoundException(id));
+
+        userService.unlikeIngredient(user, ingredient);
     }
 
     @Transactional
-    public boolean addAllergen(Long id) {
-        Optional<User> userOptional = userService
-            .getUserByUsername(authService.getCurrentUsername());
-        if (userOptional.isPresent()) {
-            User user = userOptional.get();
-            Optional<Ingredient> ingredientOptional = ingredientRepository.findById(id);
-            if (ingredientOptional.isPresent()) {
-                Ingredient ingredient = ingredientOptional.get();
-                userService.addAllergen(user, ingredient);
-                return true;
-            }
-        }
-        return false;
+    public void addAllergen(long id, User user) throws IngredientNotFoundException {
+        Ingredient ingredient = ingredientRepository.findById(id)
```

Comment:
Use `getIngredientById` instead, also wouldn't it be more clean to have `addAllergen` only in the user domain?

Reply:
> wouldn't it be more clean to have `addAllergen` only in the user domain?

yes, but that isn't part of this task, it's part of #124

Reply:
#124 involves gathering all user-related concepts and endpoints into user domain

Reply:
okay